### PR TITLE
Credits for 8.10

### PR DESCRIPTION
**Pull request by mattam82**
Work in progress. I might have no connectivity for a while so putting it up there.

### CODE REVIEW

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]:
```suggestion
  contained in the deprecation warning emitted by the previous version of Coq. For projects that require
```

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]: This link could now be: https://coq.inria.fr/distrib/V8.9.0/refman/proof-engine/detailed-tactic-examples.html#quote

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]: I think this is more appropriate in an `Ltac` category than in `Tactics`...

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]: If we decide to have a "Main changes" / "Other changes" separation, then the note about `Unset Simplex` should move to the previous section.

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]:
```suggestion
  `core` database (as previously), but a deprecation warning is emitted.
```

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]:
```suggestion
- The syntax of the :tacn:`autoapply` tactic was fixed to conform with preexisting
```

File: doc/sphinx/changes.rst
Comment (ejgallego) [reply doc/sphinx/changes.rst]: Thanks @Zimmi48 , I'd write "which is scheduled for deprecation" instead of the removal. Also, worth adding could be "the Dune build system provides an improved experience for plugin developers".

File: doc/sphinx/changes.rst
Comment (ejgallego) [inline doc/sphinx/changes.rst]: Worth mentioning on the main list of changes is that the CI system now tests Coq against OCaml trunk. This is IMO worth to be advertised more widely to OCaml people.

File: doc/sphinx/changes.rst
Comment (ejgallego) [reply doc/sphinx/changes.rst]: This should definitively be bumped to the main changes section.

File: doc/sphinx/changes.rst
Comment (ejgallego) [reply doc/sphinx/changes.rst]: I'm all for it, and also solves the problem of crediting reviewers as it just one click away.

**Comment by ejgallego** [comment]
Nice! A couple of comments:
- I see there is no documentation section, maybe you would like to note that we now distribute "drafty" API docs thanks to `odoc`.
- The Coqlib registration changes seem to be missing.
The rest looks pretty good!

**Comment by ejgallego** [comment]
As a curiosity this is the current ouput of `git shortlog -n -s --no-merges  V8.10+alpha..HEAD`
```
   299  Gaëtan Gilbert
   242  Emilio Jesus Gallego Arias
   149  Hugo Herbelin
   118  Théo Zimmermann
   114  Pierre-Marie Pédrot
   113  Vincent Laporte
    92  Enrico Tassi
    90  Maxime Dénès
    49  Jason Gross
    45  Yves Bertot
    42  Matthieu Sozeau
    41  Tanaka Akira
    35  Clément Pit-Claudel
    27  Jim Fehrle
    19  Michael Soegtrop
    15  Pierre Roux
    12  Arthur Charguéraud
    10  Jasper Hugunin
     9  Guillaume Melquiond
     7  Yishuai Li
     5  Samuel Gruetter
     5  Vincent Semeria
     5  Xia Li-yao
     4  Tej Chajed
     4  whitequark
     3  Andres Erbsen
     3  Joachim Breitner
     3  Matej Košík
     3  Sam Pablo Kuper
     2  Chantal Keller
     2  Christian Doczkal
     2  Cyril Cohen
     2  Erik Martin-Dorel
     2  Frédéric Besson
     2  Gan Shen
     2  Kayla Ngan
     2  Kazuhiko Sakaguchi
     2  Nick Lewycky
     2  thery
     1  Assia Mahboubi
     1  Benjamin Barenblat
     1  David A. Dalrymple
     1  Frédéric Chapoton
     1  Kamil Trzciński
     1  Karl Palmskog
     1  Lasse Blaauwbroek
     1  Lukasz Czajka
     1  Martin Bodin
     1  Matěj Grabovský
     1  Mirai IKEBUCHI
     1  Olivier Laurent
     1  Ryan Scott
     1  Simon Gregersen
     1  Simon Marechal
     1  Thierry Martinez
     1  Théo Winterhalter
     1  Yao Li
     1  beta
     1  hawnzug
     1  llee454@gmail.com
     1  nlewycky
     1  soraros
     1  wkwkes
     1  yudetamago
```

File: doc/sphinx/changes.rst
Comment (ejgallego) [reply doc/sphinx/changes.rst]: We should also credit @rgrinberg here and thank the Dune team.

File: doc/sphinx/changes.rst
Comment (gares) [reply doc/sphinx/changes.rst]: Yes, I'd credit @garrigue here as well.

File: doc/sphinx/changes.rst
Comment (gares) [inline doc/sphinx/changes.rst]: This is already said before.

File: doc/sphinx/changes.rst
Comment (gares) [reply doc/sphinx/changes.rst]: What do you mean Theo? (We can't say, see the doc section 4.2, since numbers are gone)

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Indeed, I realized afterwards, fixed

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: I don't see how to phrase it, but I've added @garrigue here.

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: @gares how would you say it?

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: I used basically the command shown by Emilio

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: I'd keep the detailed changes instead to avoid scattering info, even if it costs a bit of duplication.

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Don't have time to do it now, feel free to add credits to the detailed changes

File: doc/sphinx/changes.rst
Comment (Zimmi48) [reply doc/sphinx/changes.rst]: OK. Let's try to find a way to enhance this command to include co-authors.

File: doc/sphinx/changes.rst
Comment (Zimmi48) [reply doc/sphinx/changes.rst]: OK.

File: doc/sphinx/changes.rst
Comment (Zimmi48) [reply doc/sphinx/changes.rst]: We can do:

```rst
See Section :ref:`introduction-in-the-context`.
```

File: doc/sphinx/changes.rst
Comment (Zimmi48) [reply doc/sphinx/changes.rst]: Yes, that's the whole question of “Summary of changes” + “Detailed changes” or “Main changes” + “Other changes”. @mattam82 expressed how he preferred a consistent view in “Detailed changes” at the price of duplication (that's what we had before with the `CHANGES` file).

File: doc/sphinx/changes.rst
Comment (Zimmi48) [inline doc/sphinx/changes.rst]:
```suggestion
|Coq| 8.9. Vincent Laporte is the release manager and maintainer of this
```

**Comment by Zimmi48** [comment]
>Don't have time to do it now, feel free to add credits to the detailed changes

I might try to do that but probably not before next week, unless I can't help but procrastinate the many things I have to do this week.

File: doc/sphinx/changes.rst
Comment (vbgl) [reply doc/sphinx/changes.rst]: The feature has just been ported to 8.10.

**Comment by Zimmi48** [comment]
I'm giving this PR a go.

**Comment by Zimmi48** [comment]
@mattam82 Looks like you actually unchecked the box allowing others to commit to this branch.

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Unless you have time to do it Theo I would rather suggest to make it a requirement for entries in changes in the future.

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Right

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: I let you reorganize as you see fit, I can’t do complicated edits from my iPad here :)

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Right

File: doc/sphinx/changes.rst
Comment (mattam82) [reply doc/sphinx/changes.rst]: Indeed

**Comment by mattam82** [comment]
@Zimmi48 I committed your suggestions and added you as a collaborator on my repo, so you should be able to push changes now

**Comment by Zimmi48** [comment]
Thanks. I might have a go towards the end of the day, but I'll let you know if that's the case.

**Comment by Zimmi48** [comment]
I'm giving it a go now.

**Comment by Zimmi48** [comment]
I've rebased, and pushed a commit experimenting with systematic listing of PR numbers and authors in parenthesis, in the style of Dune's changes.

**Comment by mattam82** [comment]
This is looking good!

**Comment by Zimmi48** [comment]
I've finished adding links to PRs and names of authors for all change entries and I've refactored the structure to avoid the duplication of entries: https://coq.gitlab.io/-/coq/-/jobs/201416869/artifacts/_install_ci/share/doc/coq/sphinx/html/changes.html

**Comment by vbgl** [comment]
@Zimmi48 @mattam82 Will Théo be able to finish the work of this PR or is some help from Matthieu needed? Thanks.

**Comment by mattam82** [comment]
I think it’s ready, leaving it to Théo to polish the last things as I’m leaving for the Bolivian Amazon ;) Thanks again Théo for the uniformisation and proper credits!

**Comment by Zimmi48** [comment]
I can do a final pass to take into accounts the last remarks that were made in this thread by various people. There are also three holes left by @mattam82 about closed issues, merged PRs and number of commits, that will get more accurate the closer to the release we are (and BTW, I don't know how to get the number of commits). @vbgl When do you need this?

**Comment by vbgl** [comment]
> When do you need this?

Monday should be fine. Thank you.

**Comment by Zimmi48** [comment]
>I see there is no documentation section, maybe you would like to note that we now distribute "drafty" API docs thanks to odoc.

WDYT @vbgl? Should we start advertizing https://coq.github.io/doc/master/api, https://coq.github.io/doc/master/refman and https://coq.github.io/doc/master/stdlib, or should we wait for the end of the transfer of the website to GitHub pages so that these URLs are actually starting with https://coq.inria.fr/?

**Comment by vbgl** [comment]
I guess that the coq.github.io urls will stay for a while. It seems good to advertise them already.

**Comment by Zimmi48** [comment]
As far as I'm concerned, this is ready to go. Careful reviews are still welcome though. If more PRs with CHANGES entry are still intended to be merged after this one in the 8.10+beta1 milestone (e.g. #9996), then their authors should move the entry in the Changes chapter. Note that this is a bit artisanal but should be handled automatically by the next release.

**Review by vbgl — APPROVED** [review]

**Comment by Zimmi48** [comment]
Rebased after #9995, first 8.11 PR with a CHANGES entry. Hopefully, this is the last rebase.

**Comment by vbgl** [comment]
Thank you Théo. Do I need to wait for CI completion before merging?

**Comment by Zimmi48** [comment]
Last pipeline (before rebase) was successful: https://gitlab.com/coq/coq/pipelines/58989127
My rebase only changed the content of `CHANGES.md` so *a priori* there is no risk in merging now.

**Pull request merged**
